File: algorithm/tree/symmetricTree/symmetricTree.py
#!/usr/bin/python
#
#

def isSymmetric(self, root):
  if root is None:
    return True
  return self.isSym(root.left, root.right)


def isSym(self, l, r):
  if l is None and r is None:
    return True
  if l is None or r is None:
    return False
  if l.val != r.val
    return False
  return self.isSym(l.left, r.right) and self.isSym(l.right, r.left)

File: algorithm/tree/symmetricTree/symmetricTree.java
public class symmetricTree {

  publc boolean isSymmetric(TreeNode root) {
    if (root == null) return true;
    return isSym(root.left, root.right);
  }

  public boolean isSym(TreeNode l, TreeNode r) {
    if (l == null && r == null) {
      return true;
    }else if (l == null || r == null) {
      return false;
    }

    if (l.val != r.val) {
      return false;
    }

    return isSym(l.left, r.right) && isSym(l.right, r.left);
  }
}
